KeeAgent/KeeAgentInteractiveUi.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Threading;
+using System.Windows.Forms;
+
+namespace KeeAgent
+{
+  /// <summary>
+  /// Provides a separate UI thread for interactive communication without interlocking with the MainWindow.
+  /// </summary>
+  public sealed class KeeAgentInteractiveUi : IDisposable
```

Comment:
nitpick: `Interactive` isn't meaningful to me in this context. Could we call it something else?

```suggestion
  public sealed class KeeAgentUiThread : IDisposable
```

Reply:
@dlech , done